lib/usb/usb_f107.c

```diff
@@ -52,21 +52,31 @@ const struct _usbd_driver stm32f107_usb_driver = {
 /** Initialize the USB device controller hardware of the STM32. */
 static usbd_device *stm32f107_usbd_init(void)
 {
-	OTG_FS_GINTSTS = OTG_GINTSTS_MMIS;
-
 	OTG_FS_GUSBCFG |= OTG_GUSBCFG_PHYSEL;
-	/* Enable VBUS sensing in device mode and power down the PHY. */
-	OTG_FS_GCCFG |= OTG_GCCFG_VBUSBSEN | OTG_GCCFG_PWRDWN;
 
 	/* Wait for AHB idle. */
 	while (!(OTG_FS_GRSTCTL & OTG_GRSTCTL_AHBIDL));
 	/* Do core soft reset. */
 	OTG_FS_GRSTCTL |= OTG_GRSTCTL_CSRST;
 	while (OTG_FS_GRSTCTL & OTG_GRSTCTL_CSRST);
 
+	uint32_t id = OTG_FS_CID;
+	if (id == OTG_CID_HAS_VBDEN) {
+		/* Enable VBUS detection in device mode and power up the PHY. */
+		OTG_FS_GCCFG |= OTG_GCCFG_VBDEN | OTG_GCCFG_PWRDWN;
+		/* Set the Soft Connect (STMF32446, STMF32469 comes up disconnected) */
+		OTG_FS_DCTL &= ~OTG_DCTL_SDIS;
```

Comment:
The this line of code makes up for the microcode change to the Poweron state of this bit. With this change the F469 and F446 have the Soft Disconnect state the same as the F427 at power up. Which is connected. It would 
NOT be appropriate for the F427